go/vt/mysqlctl/backup_blackbox_race_test.go

```diff
@@ -0,0 +1,152 @@
+//go:build !race
```

Comment:
After working on this test I discovered a data race in how we write/flush our backup files, I have opened an issue to document the issue https://github.com/vitessio/vitess/issues/17065, but it does not seem to be cause by the change I made in this PR. The `f.Chmod(0444)` done in the test exposes the issue, without it, the race is not triggered.

I could work on a follow-up and fix the race condition in another PR, in the meantime I have moved this test to a file that we do not run when `go test -race` is ran.

Link to a GitHub Action showing the failure: https://github.com/vitessio/vitess/actions/runs/11507118183?pr=17062

Reply:
We should try to fix this data race before we do patch releases again.